packages/indicators/src/builders/buildArithmetic.ts

```diff
@@ -3,44 +3,84 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import { getVariables, runArithmetic } from '@beyondessential/arithmetic';
+
 import { Aggregator } from '@tupaia/aggregator';
-import {
-  getAggregationsByCode,
-  extractDataElementCodesFromFormula,
-  fetchAnalytics,
-} from './helpers';
-import { Builder, AggregationSpecs, FetchOptions } from '../types';
-
-interface ArithmeticConfig {
+import { hasContent, isAString, isPlainObject } from '@tupaia/utils';
+import { getAggregationsByCode, fetchAnalytics, validateConfig } from './helpers';
+import { Analytic, Builder, AggregationSpecs, FetchOptions } from '../types';
+
+export interface ArithmeticConfig {
+  readonly [key: string]: unknown;
   readonly formula: string;
   readonly aggregation: AggregationSpecs;
 }
 
-const validateConfig = (config: ArithmeticConfig) => {
-  const { formula, aggregation } = config;
-  const codesInFormula = extractDataElementCodesFromFormula(formula);
+interface AnalyticalEvent {
+  organisationUnit: Analytic['organisationUnit'];
+  period: Analytic['period'];
+  dataValues: Record<Analytic['dataElement'], Analytic['value']>;
+}
 
-  codesInFormula.forEach(codeInFormula => {
-    if (!Object.keys(aggregation).includes(codeInFormula)) {
-      throw new Error(
-        `Data element ${codeInFormula} is referenced in the formula but has no aggregation defined`,
-      );
+const assertAggregationIsDefinedForCodesInFormula = (
+  aggregation: string,
+  { formula }: { formula: ArithmeticConfig['formula'] },
+) => {
+  getVariables(formula).forEach(code => {
+    if (!Object.keys(aggregation).includes(code)) {
+      throw new Error(`'${code}' is referenced in the formula but has no aggregation defined`);
     }
   });
 };
 
-export const buildArithmetic: Builder<ArithmeticConfig> = async (input: {
-  aggregator: Aggregator;
-  config: ArithmeticConfig;
-  fetchOptions: FetchOptions;
-}) => {
-  const { aggregator, config, fetchOptions } = input;
-  validateConfig(config);
+const configValidators = {
+  formula: [hasContent, isAString],
+  aggregation: [hasContent, isPlainObject, assertAggregationIsDefinedForCodesInFormula],
+};
```

Comment:
Typescript types made me go all the way with config validation, since I wasn't really asserting that the passed `config` in this builder is of an `ArithmeticConfig` type. Initially I thought that this may be an overkill, but eventually I reckon it's actually a good thing: using `ObjectValidator` + `validatorFunctions.js` from `@tupaia/utils` makes this pretty clean, I hope.

One caveat using those is captured in this card: [1201: Generalise object validation errors](https://github.com/beyondessential/tupaia-backlog/issues/1201)